Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
Ooooooooh I finally understood. You want every section to have an index 0 because each section controller is some sort of auto-sufficient unit that has no knowledge about the other sections, so if I'm a section controller and I'm implementing my own supplementary view source, I should know everything I need for my own view, without depending on other sections. I see what you mean, this is good!

But do you see now that it also makes sense if the section controller knows its section index? I thought this was possible and intended to be possible from the index parameter that is being passed to the supplementary view source. However, now I need to go back to my question, you answered it before but I didn't understand a practical usage. What is this index parameter for in your current architecture? Where do you use it in Instagram for example?